### ISSUE TITLE
make release 6.2.0

### ISSUE THREAD

**Issue by alifeee** [burnash/gspread#1516]
here is the auto-generated changelog at the moment

## What's Changed
* Add property expiry in gspread client by lavigne958 in https://github.com/burnash/gspread/pull/1453
* Bump typing-extensions from 4.11.0 to 4.12.0 by dependabot in https://github.com/burnash/gspread/pull/1471
* Fix code block formatting typo in README by agrvz in https://github.com/burnash/gspread/pull/1474
* ignore jinja CVE by lavigne958 in https://github.com/burnash/gspread/pull/1481
* Type part of test suite utils by lavigne958 in https://github.com/burnash/gspread/pull/1483
* Remove passing exception as args to super in APIError by mike-flowers-airbnb in https://github.com/burnash/gspread/pull/1477
* Bump mypy from 1.10.0 to 1.10.1 by dependabot in https://github.com/burnash/gspread/pull/1488
* Update advanced.rst by yatender-rjliving in https://github.com/burnash/gspread/pull/1492
* Bump bandit from 1.7.8 to 1.7.9 by dependabot in https://github.com/burnash/gspread/pull/1485
* Bump flake8 from 7.0.0 to 7.1.0 by dependabot in https://github.com/burnash/gspread/pull/1486
* Bump typing-extensions from 4.12.0 to 4.12.2 by dependabot in https://github.com/burnash/gspread/pull/1480
* Bump mypy from 1.10.1 to 1.11.1 by dependabot in https://github.com/burnash/gspread/pull/1497
* Bump black from 24.4.2 to 24.8.0 by dependabot in https://github.com/burnash/gspread/pull/1499
* Bump flake8 from 7.1.0 to 7.1.1 by dependabot in https://github.com/burnash/gspread/pull/1501
* Fix docstring about BackOffHTTPClient by pataiji in https://github.com/burnash/gspread/pull/1502
* Fix comment to reflect correct google-auth package version requirement by ikmals in https://github.com/burnash/gspread/pull/1503
* Doc/community addons orm package by lavigne958 in https://github.com/burnash/gspread/pull/1506
* fix: fix type annotation for default_blank by hiro-o918 in https://github.com/burnash/gspread/pull/1505
* Bump mypy from 1.11.1 to 1.11.2 by dependabot in https://github.com/burnash/gspread/pull/1508
* better handler API error parsing. by lavigne958 in https://github.com/burnash/gspread/pull/1510
* Add test on receiving an invalid JSON in the APIError exception handler. by lavigne958 in https://github.com/burnash/gspread/pull/1512
* [feature] Add 'expand_table' feature by lavigne958 in https://github.com/burnash/gspread/pull/1475
* Bump bandit from 1.7.9 to 1.7.10 by dependabot in https://github.com/burnash/gspread/pull/1514
* Created a `batch_merge` function [Issue #1473] by muddi900 in https://github.com/burnash/gspread/pull/1498
* Added a range option to `Worksheet.get_notes` [Issue #1482] by muddi900 in https://github.com/burnash/gspread/pull/1487
* Documentation update for gspread.worksheet.Worksheet.get_all_records by levon003 in https://github.com/burnash/gspread/pull/1529
* add example for `batch_merge` by alifeee in https://github.com/burnash/gspread/pull/1542

## New Contributors
* agrvz made their first contribution in https://github.com/burnash/gspread/pull/1474
* mike-flowers-airbnb made their first contribution in https://github.com/burnash/gspread/pull/1477
* yatender-rjliving made their first contribution in https://github.com/burnash/gspread/pull/1492
* pataiji made their first contribution in https://github.com/burnash/gspread/pull/1502
* ikmals made their first contribution in https://github.com/burnash/gspread/pull/1503
* hiro-o918 made their first contribution in https://github.com/burnash/gspread/pull/1505
* levon003 made their first contribution in https://github.com/burnash/gspread/pull/1529

**Full Changelog**: https://github.com/burnash/gspread/compare/v6.1.2...v6.2.0
should this be a minor release? are there any non-backwards compatible changes

otherwise, we could make 6.2.0

**Comment by the4thamigo-uk** [comment]
Do you have a rough ETA for when the release will be available? We are suffering from https://github.com/burnash/gspread/pull/1512 at the moment.

**Comment by lavigne958** [comment]
Not yet. On my side. We need to check what has been planned and won't be in the release, what has been planned and *must* be in the release and then make it.

**Comment by K7CTC** [comment]
First, thank you for such an amazing Python package!  We are also suffering from #1512 at the moment.  Started on 2024-09-18 for us.  Looking forward to the next release!

**Comment by JoachimHubele** [comment]
In case you are suffering from the JSON APIError (#1510, #1512) like me, here is a potential workaround until release 6.2.0 is out. 

If you can run your code in an virtual environment, replace the file exceptions.py in the  sitepackages/gspread with the one from the Pull Request: https://github.com/burnash/gspread/pull/1510/commits. Worked for me.

**Comment by john-ramsey** [comment]
We are dealing with the above as well #1510 and #1512 as well - do we have a timeline on making the latest release?

**Comment by eitchtee** [comment]
The intermittent 502 errors seems to be a bug on Google's end, related issue: https://issuetracker.google.com/issues/369670473

**Comment by K7CTC** [comment]
> The intermittent 502 errors seems to be a bug on Google's end, related issue: https://issuetracker.google.com/issues/369670473

Yes, this exactly!  Thank you @eitchtee for the link, this is useful to know.

**Comment by alifeee** [comment]
hi all

a new release `6.1.3` has been released ([on github](https://github.com/burnash/gspread/releases/tag/v6.1.3)) ([on PyPi](https://pypi.org/project/gspread/6.1.3/)) which contains a fix for this issue

we now catch the non-JSON error and turn it into a `GSpreadException` which should be easier to catch and find out what went wrong with the request

see more context here: https://github.com/burnash/gspread/pull/1510

**Comment by alifeee** [comment]
@lavigne958 we could make this release as a minor feature release soon

**Comment by alifeee** [comment]
I have made this release
